addons/stock/static/src/views/list/inventory_report_list_controller.js

```diff
@@ -0,0 +1,115 @@
+/** @odoo-module */
+
+import { ListController } from '@web/views/list/list_controller';
+import { useBus } from "@web/core/utils/hooks";
+import { session } from "@web/session";
+
+export class InventoryReportListController extends ListController {
+
+    setup() {
+        super.setup();
+        if (this.props.context.inventory_mode || this.props.context.inventory_report_mode) {
+            useBus(this.model, "record-updated",
+                this.recordUpdated
+            );
+        }
+    }
+
+    /**
+     * Handler called when the user clicked on the 'Valuation at Date' button.
+     * Opens wizard to display, at choice, the products inventory or a computed
+     * inventory at a given date.
+     */
+    async onClickInventoryAtDate() {
+        var context = {
+            active_model: this.props.resModel,
+        };
+        if (this.props.context.default_product_id) {
+            context.product_id =  this.props.context.default_product_id;
+        }
+        else if (this.props.context.product_tmpl_id) {
+            context.product_tmpl_id =  this.props.context.product_tmpl_id;
+        }
+        this.actionService.doAction({
+            res_model: 'stock.quantity.history',
+            views: [[false, 'form']],
+            target: 'new',
+            type: 'ir.actions.act_window',
+            context: context,
+        });
+    }
+
+    /**
+     * Count 'resId' loaded records.
+     * Multiple records occurs when adding an already loaded-in-view record.
+     */
+    _countResIdRecords(level, resId) {
+        let count = 0;
+        if (level.groupByInfo) {
+            if (level.isGrouped) {
+                for (const i in level.groups) {
+                    count += this._countResIdRecords(level.groups[i], resId);
+                }
+            }
+            else {
+                for (const record of level.list.records) {
+                    if (record.resId == resId) {
+                        count++;
+                    }
+                }
+            }
+        }
+        else {
+            for (const record of level.records) {
+                if (record.resId == resId) {
+                    count++;
+                }
+            }
+        }
```

Comment:
To avoid the repetition:
```suggestion
        const records = level.groupByInfo ? level.list.records : level.records;
        if (level.groupByInfo && level.isGrouped) {
            for (const i in level.groups) {
                count += this._countResIdRecords(level.groups[i], resId);
            }
        } else {
            count += records.filter(record => record.res_id == resId).lenght
            }
        }
```
You can also use `filter` to find the records with the same `res_id` and increment `count` by the filter's result's lenght (it's shorter :) )